### DISCUSSION TITLE
not receiving foundry.toml

### DISCUSSION THREAD

**Discussion by khandelwalmoksh** [Cyfrin/foundry-full-course-cu#3052] [Q&A] [answered]
![Screenshot (34)](https://github.com/user-attachments/assets/eb56863b-6c19-48d0-9af9-22bebe746e25)
even though installed same version as suggested by Patrick 
```forge install smartcontractkit/chainlink-brownie-contracts@1.1.1 --no-commit```

**Comment by EngrPips** [comment, marked as answer]
Hello @khandelwalmoksh, You need to initialize a Foundry project to get a `Foundry.toml` file. You can do that by running `forge init` inside of the Folder you want the Foundry project initiated.

**Comment by khandelwalmoksh** [comment]
now, this file is not there
![Screenshot (35)](https://github.com/user-attachments/assets/b4c5c7f8-9295-489b-9573-56767d80eb42)

**Reply by EngrPips** [reply]
Hello @khandelwalmoksh, I think your import command should be the below 

```js
import {VRFConsumerBaseV2Plus} from "@chainlink/contracts/src/v0.8/vrf/dev/VRFConsumerBaseV2Plus.sol";
```

**Comment by khandelwalmoksh** [comment]
still an error here
in file, check the files i.e. the left side. It might help you
![Screenshot (36)](https://github.com/user-attachments/assets/dbd4e57f-883f-411d-9931-958b7a0010ee)

**Reply by EngrPips** [reply]
Alright, use `cltr + p` [Windows] or `command + p` [Mac] and type `VRFConsumerBaseV2Plus` in the search bar that appears, and let's see if the file exists in your codebase or not.

**Reply by khandelwalmoksh** [reply]
Actually, I added ```VRFConsumerBaseV2Plus``` manually with help of chatgpt. Now, I am getting different errors 😁
![Screenshot (37)](https://github.com/user-attachments/assets/473b23e1-de00-4e80-a85a-7e96b2d053a8)

**Reply by EngrPips** [reply]
run `forge install smartcontractkit/chainlink-brownie-contracts@0.6.1 --no-commit` in your terminal then retry the compilation.

**Reply by khandelwalmoksh** [reply]
output ```fatal: 'foundry-fund-me-f23/foundry-smart-contract-lottery-cu/lib/chainlink-brownie-contracts' already exists and is not a valid git repo
Error: git submodule exited with code 128```

**Reply by EngrPips** [reply]
open you `chainlink-brownie-contracts` folder and show what is there.

**Reply by khandelwalmoksh** [reply]
![Screenshot (38)](https://github.com/user-attachments/assets/7d6984ba-a884-434d-8c4d-6eb88713aaa7)

**Reply by EngrPips** [reply]
> Actually, I added `VRFConsumerBaseV2Plus` manually with help of chatgpt. Now, I am getting different errors 😁 ![Screenshot (37)](https://private-user-images.githubusercontent.com/149714354/396488416-473b23e1-de00-4e80-a85a-7e96b2d053a8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ0NTk5OTksIm5iZiI6MTczNDQ1OTY5OSwicGF0aCI6Ii8xNDk3MTQzNTQvMzk2NDg4NDE2LTQ3M2IyM2UxLWRlMDAtNGU4MC1hODVhLTdlOTZiMmQwNTNhOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxN1QxODIxMzlaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xN2EyNmNmNGQyODc4MTIzYjc4MGYxZWM1MjZjMDUzM2JjNzVjNzNlYTgyYTJlNmIyY2Q5ZDdlMzBlOWMzNGY2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.XvqHjisddh3tb3yHCLpHprpVCx5no5pmQMgNnl2u-0k)

I am not sure if that is the correct package that you have installed. Delete that and run this in your terminal `forge install smartcontractkit/chainlink-brownie-contracts --no-commit`

**Reply by khandelwalmoksh** [reply]
I re-cloned the whole project, done ```forge init``` & ```forge install smartcontractkit/chainlink-brownie-contracts --no-commit```. Stil this error, even though the necessary files are present
![Screenshot (39)](https://github.com/user-attachments/assets/eb80ba0d-121a-4943-baeb-f4a8e32d1d3a)

It might be my system couldn't find the file. Earlier, an import was giving error, after 4 days it automatically found it and error was gone

**Reply by EngrPips** [reply]
Is the error gone now.

**Reply by khandelwalmoksh** [reply]
no, still exists

**Reply by EngrPips** [reply]
Are you sure you pointed to the correct path of the file @khandelwalmoksh

**Reply by khandelwalmoksh** [reply]
yes

**Reply by EngrPips** [reply]
I am unsure what could be the problem. What lesson are you currently watching?

**Reply by khandelwalmoksh** [reply]
https://updraft.cyfrin.io/courses/foundry/smart-contract-lottery/implementing-chainlink-vrf?lesson_format=video watch from 6:55
or  Foundry Fundamentals>Section 4  7. Implement the Chainlink VRF
here, ```forge install smartcontractkit/chainlink-brownie-contracts@1.1.1 --no-commit``` is used

**Reply by EngrPips** [reply]
Did you follow the installation and remapping guides on this page https://github.com/smartcontractkit/chainlink-brownie-contracts to install and set remappings for the `chainlink-brownie-contracts`?

**Reply by khandelwalmoksh** [reply]
yes, exactly the same. Infact, copied remapping from the ReadMe from Github. There are all the required files present which are seen in the lecture
I think, I should leave it for now and proceed. Will come back later

**Reply by EngrPips** [reply]
Alright, If you come back to it and still don't achieve any success fixing it, Please tag me and send the GitHub link of your project so that I can try to set it up locally on my end.

**Reply by khandelwalmoksh** [reply]
hey @EngrPips, I deleted & re-installed everything. SUCCESSS!  I strongly believe its from my system's side or might be environment...

**Reply by EngrPips** [reply]
> hey @EngrPips, I deleted & re-installed everything. SUCCESSS! I strongly believe its from my system's side or might be environment...

Awesome, I am glad you figured it out.
